File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterSyncQueriesStartedEvent.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.hub

import org.json.JSONArray

class FlutterSyncQueriesStartedEvent(override val eventName: String, private val models: Array<String>) :
    FlutterHubEvent {
    override fun toValueMap(): Map<String, Any> {
        return mapOf(
            "eventName" to this.eventName,
            "models" to JSONArray(this.models).toString()
        )
    }
}

File: packages/amplify_datastore/ios/Classes/types/hub/FlutterSyncQueriesStartedEvent.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation
import Amplify
import AmplifyPlugins

struct FlutterSyncQueriesStartedEvent: FlutterHubEvent {
    var eventName: String
    var models: [String]

    init(payload: HubPayload) throws {
        guard let syncQueriesStarted = payload.data as? SyncQueriesStartedEvent else {
                  throw FlutterDataStoreError.hubEventCast
              }
        self.eventName = shortEventName(eventName: payload.eventName)
        self.models = syncQueriesStarted.models
    }

    func toValueMap() -> [String: Any] {
        return [
            "eventName": eventName,
            "models": models.description
        ]
    }
}
